compiler/parser/chapel.ypp

```diff
@@ -480,7 +480,8 @@ stmt:
 | TCONTINUE opt_ident TSEMI            { $$ = buildGotoStmt(GOTO_CONTINUE, $2); }
 | TDELETE expr TSEMI                   { $$ = buildPrimitiveStmt(PRIM_DELETE, $2); }
 | TLABEL ident stmt                    { $$ = buildLabelStmt($2, $3); }
-| TLOCAL stmt                          { $$ = buildLocalStmt($2); }
+| TLOCAL expr do_stmt                  { $$ = buildLocalStmt($2, $3); }
+| TLOCAL do_stmt                       { $$ = buildLocalStmt(new SymExpr(gTrue), $2); }
```

Comment:
Continuing our conversation in the top-level notes, I think I'd expect this to simply be `buildLocalStmt($2)` in order to avoid the early-compiler code bloat that you noted.